### PR TITLE
Add title section to customization page

### PR DESCRIPTION
**Pull request by palewire**
For #2748 and #2195

I also added the same example to the Iowa case study

### CODE REVIEW

File: doc/user_guide/customization.rst [hunk 1]
```diff
@@ -121,6 +121,46 @@ Global configurations should be reserved for creating themes that are applied
...
+       color="source:N"
+   )
+
+It is also possible to add a subtitle, and `configure <https://altair-viz.github.io/user_guide/configuration.html#config-title>`_ a number of other attributes, by passing in an `alt.Title` object.
+
+.. altair-plot::
+
+   alt.Chart(
+      iowa,
+      title=alt.Title("Iowa's green energy boom", subtitle="A growing share of the state's energy has come from renewable sources")
```
Comment (joelostblom) [inline doc/user_guide/customization.rst:157]:
```suggestion
      title=alt.Title(
          "Iowa's green energy boom",
          subtitle="A growing share of the state's energy has come from renewable sources"
      )
```

**Review by joelostblom — CHANGES_REQUESTED** [review]
Thanks @palewire ! Could you add the two last examples from https://stackoverflow.com/questions/57244390/how-to-add-a-subtitle-to-an-altair-generated-chart/66147967#66147967 as well?

**Comment by palewire** [comment]
Okay. I took a run at that.

File: doc/user_guide/customization.rst [hunk 2]
```diff
@@ -121,6 +121,85 @@ Global configurations should be reserved for creating themes that are applied
...
+       color="source:N"
+   )
+
+It is also possible to add a subtitle by passing in an `alt.TitleParams` object.
+
+.. altair-plot::
+
+   alt.Chart(
+      iowa,
+      title=alt.TitleParams(
```
Comment (joelostblom) [inline doc/user_guide/customization.rst:157]:
```suggestion
      title=alt.Title(
```

File: doc/user_guide/customization.rst [hunk 3]
```diff
@@ -121,6 +121,85 @@ Global configurations should be reserved for creating themes that are applied
...
+          "Iowa's green energy boom",
+          subtitle="A growing share of the state's energy has come from renewable sources"
+      )
+   ).mark_area().encode(
+       x="year:T",
+       y=alt.Y("net_generation:Q", stack="normalize"),
+       color="source:N"
+   )
+
+The subtitle can run to two lines.
```
Comment (joelostblom) [inline doc/user_guide/customization.rst:167]:
```suggestion
The subtitle can run to two lines by passing a list where each list item is a line (if you don't want to create this list manually as in the example below, you can use the ``wrap`` function from the `textwrap library https://docs.python.org/3/library/textwrap.html`_ to split a string into a list of substrings of a certain length).
```

File: doc/user_guide/customization.rst [hunk 4]
```diff
@@ -121,6 +121,85 @@ Global configurations should be reserved for creating themes that are applied
...
+
+.. altair-plot::
+
+   alt.Chart(
+      iowa,
+      title=alt.TitleParams(
```
Comment (joelostblom) [inline doc/user_guide/customization.rst:173]:
```suggestion
      title=alt.Title(
```

File: doc/user_guide/customization.rst [hunk 5]
```diff
@@ -121,6 +121,85 @@ Global configurations should be reserved for creating themes that are applied
...
+       color="source:N"
+   )
+
+The parameters object can also `configure <https://altair-viz.github.io/user_guide/configuration.html#config-title>`_ a number of other attributes.
+
+.. altair-plot::
+
+   alt.Chart(
+      iowa,
+      title=alt.TitleParams(
```
Comment (joelostblom) [inline doc/user_guide/customization.rst:189]:
```suggestion
      title=alt.Title(
```

File: tests/examples_methods_syntax/iowa_electricity.py [hunk 6]
```diff
@@ -9,7 +9,10 @@
 
 source = data.iowa_electricity()
 
-alt.Chart(source, title="Iowa's renewable energy boom").mark_area().encode(
+alt.Chart(
+    source,
+    title=alt.TitleParams("Iowa's green energy boom", subtitle="A growing share of the state's energy has come from renewable sources")
```
Comment (joelostblom) [inline tests/examples_methods_syntax/iowa_electricity.py:14]:
```suggestion
    title=alt.Title(
        "Iowa's green energy boom",
        subtitle="A growing share of the state's energy has come from renewable sources"
    )
```

File: doc/user_guide/customization.rst [hunk 7]
```diff
@@ -121,6 +121,85 @@ Global configurations should be reserved for creating themes that are applied
...
+          "Iowa's green energy boom",
+          subtitle=["A growing share of the state's energy", "has come from renewable sources"]
+      )
+   ).mark_area().encode(
+       x="year:T",
+       y=alt.Y("net_generation:Q", stack="normalize"),
+       color="source:N"
+   )
+
+The parameters object can also `configure <https://altair-viz.github.io/user_guide/configuration.html#config-title>`_ a number of other attributes.
```
Comment (joelostblom) [inline doc/user_guide/customization.rst:183]:
```suggestion
The ``Title`` object can also configure a number of other attributes, e.g., the position of the title and subtitle (see see :ref:`user-guide-customization` for details).
```

**Review by joelostblom — CHANGES_REQUESTED** [review]
Great, thank you!  Just a few more comments and then we can merge

**Comment by palewire** [comment]
I've accepted all your changes

**Review by joelostblom — APPROVED** [review]
Great, thank again!

**Pull request merged**